_field-types/flat-object.md

```diff
@@ -0,0 +1,218 @@
+
+## Using flat object
+
+The following example illustrates mapping a field as a flat object, indexing documents with flat object fields, and searching for leaf values of the flat object in those documents.
+
+Only the root field of a document can be defined as a flat object. You cannot define an object that is part of another JSON object as a flat object because when a flat object it is flattened to a string, the nested architecture of the leaves is lost.
+{: .note}
+
+First, create a mapping for your index, where `issue` is of type `flat_object`:
+
+```json
+PUT /test-index/
+{
+  "mappings": {
+    "properties": {
+      "issue": {
+        "type": "flat_object"
+      }
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+Next, index two documents with flat object fields:
+
+```json
+PUT /test-index/_doc/1
+{
+  "issue": {
+    "number": "123456",
+    "labels": {
+      "version": "2.1",
+      "backport": [
+        "2.0",
+        "1.3"
+      ],
+      "category": {
+        "type": "API",
+        "level": "enhancement"
+      }
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+```json
+PUT /test-index/_doc/2
+{
+  "issue": {
+    "number": "123457",
+    "labels": {
+      "version": "2.2",
+      "category": {
+        "type": "API",
+        "level": "bug"
+      }
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+To search for a leaf value of the flat object, use a POST request. Even if you don't know the field names, you can search for a leaf value in the entire flat object. For example, the following request searches for all issues labeled as bugs:
+
+```json
+POST /test-index/_search
+{
+  "query": {
+    "match": {"issue": "bug"}
+  }
+}
+```
+
+Alternatively, if you know the subfield name in which to search, provide the field's path in dot notation:
+
+```json
+POST /test-index/_search
+{
+  "query": {
+    "match": {"issue.labels.category.level": "bug"}
+  }
+}
+```
+{% include copy-curl.html %}
+
+In both previous cases, the response is the same and contains document 2:
+
+```json
+{
+  "took": 1,
+  "timed_out": false,
+  "_shards": {
+    "total": 1,
+    "successful": 1,
+    "skipped": 0,
+    "failed": 0
+  },
+  "hits": {
+    "total": {
+      "value": 1,
+      "relation": "eq"
+    },
+    "max_score": 1.0303539,
+    "hits": [
+      {
+        "_index": "test-index",
+        "_id": "2",
+        "_score": 1.0303539,
+        "_source": {
+          "issue": {
+            "number": "123457",
+            "labels": {
+              "version": "2.2",
+              "category": {
+                "type": "API",
+                "level": "bug"
+              }
+            }
+          }
+        }
+      }
+    ]
+  }
+}
+```
+
+Using a prefix query, you can search for all issues for the versions that start with `2.`:
+
+```json
+POST /test-index/_search
+{
+  "query": {
+    "prefix": {"issue.labels.version": "2."}
+  }
+}
+```
+
+With a range query, you can search for all issues for versions 2.0--2.1:
```

Comment:
Is there an extra en dash between the range?

Reply:
No, this is just an en dash in Markdown.